service/prometheus/group_test.go

```diff
@@ -0,0 +1,260 @@
+package prometheus
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+// Test_Prometheus_FilterServices tests the FilterServices function.
+func Test_Prometheus_FilterServices(t *testing.T) {
+	tests := []struct {
+		services         []v1.Service
+		expectedServices []v1.Service
+	}{
+		// Test a service without either cluster or certificate annotations is filtered.
+		{
+			services: []v1.Service{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "bar",
+					},
+				},
+			},
+			expectedServices: []v1.Service{},
+		},
+
+		// Test a service with just a cluster annotation is filtered.
+		{
+			services: []v1.Service{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "bar",
+						Annotations: map[string]string{
+							ClusterAnnotation: "xa5ly",
+						},
+					},
+				},
+			},
+			expectedServices: []v1.Service{},
+		},
+
+		// Test a service with just a certificate annotation is filtered.
+		{
+			services: []v1.Service{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "bar",
+						Annotations: map[string]string{
+							CertificateAnnotation: "default/xa5ly-prometheus",
+						},
+					},
+				},
+			},
+			expectedServices: []v1.Service{},
+		},
+
+		// Test that a service with both cluster and certificate annotations
+		// is not filtered.
+		{
+			services: []v1.Service{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "bar",
+						Annotations: map[string]string{
+							ClusterAnnotation:     "xa5ly",
+							CertificateAnnotation: "default/xa5ly-prometheus",
+						},
+					},
+				},
+			},
+			expectedServices: []v1.Service{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "bar",
+						Annotations: map[string]string{
+							ClusterAnnotation:     "xa5ly",
+							CertificateAnnotation: "default/xa5ly-prometheus",
+						},
+					},
+				},
+			},
+		},
```

Comment:
There should be the same tests for multiple services.

Reply:
Done